gst/nnstreamer/nnstreamer_plugin_api_impl.c

```diff
@@ -923,6 +923,188 @@ _peer_is_flexible_tensor_caps (GstPad * pad)
   return flexible;
 }
 
+/**
+ * @brief Check whether the tensor dimensions are same
+ */
+static gboolean
+_is_tensor_dim_same (tensor_dim dim1, tensor_dim dim2)
+{
+  guint i;
+  for (i = 0; i < NNS_TENSOR_RANK_LIMIT; i++) {
+    if (dim1[i] != dim2[i])
+      return FALSE;
+  }
+  return TRUE;
+}
+
+/**
+ * @brief Check whether two structures have the same dimension
+ */
+static gboolean
+_is_structure_dimension_same (GstStructure * structure1,
+    GstStructure * structure2)
+{
+  tensor_dim dim1, dim2;
+  const char *dim_str1;
+  const char *dim_str2;
+
+  g_return_val_if_fail (gst_structure_has_field (structure1, "dimension"),
+      FALSE);
+  g_return_val_if_fail (gst_structure_has_field (structure2, "dimension"),
+      FALSE);
+
+  dim_str1 = gst_structure_get_string (structure1, "dimension");
+  gst_tensor_parse_dimension (dim_str1, dim1);
+
+  dim_str2 = gst_structure_get_string (structure2, "dimension");
+  gst_tensor_parse_dimension (dim_str2, dim2);
+
+  if (!_is_tensor_dim_same (dim1, dim2))
```

Comment:
minor: simply `return _is_tensor_dim_same (dim1, dim2);`

Reply:
Updated.
Thank you :)